Makefile

```diff
@@ -2,7 +2,13 @@
 # Variables                                      #
 ##################################################
 SHELL           = /bin/bash
-VERSION        ?= main
+
+# VERSION is the image tag that will be used. It could be provided using environment variable IMAGE_TAG
```

Comment:
Why do we need this change? You can set environment variable VERSION to achieve the same thing.

Reply:
Are you sure? wow, I will test it today because yesterday I couldn't achieve it. They only way that I could do it is setting it during the make command, something like:
`make e2e-test VERSION=blablalba`
And the problem with it is that our flow is:
1. make e2e-test
2. run-all.sh
3. ava setup.tests.ts
4. make deploy

That's why instead of passing the variable through all the layers I thought that it's easier if we set env variables, but definitively I will try again (my solution is a bit ugly from my POV :( )

Reply:
Have you tried to set env variable `VERSION` before running make?

Reply:
I tried a lot of things so probably yes, but I will check it again because it's also probably that I did any mistake

Reply:
Maybe I'm doing something wrong, but I have just tried this:
```bash
export VERSION=test && \
echo "VERSION value: $VERSION" && \
make publish
->
VERSION value: test
docker build . -t ghcr.io/kedacore/keda:main
```
The value `main` is not replaced. But If I set explicitly, It works:
```bash
export VERSION=test && \
echo "VERSION value: $VERSION" && \
make publish VERSION=$VERSION
->
VERSION value: test
docker build . -t ghcr.io/kedacore/keda:test
```

That's why I thought to use environment variables. The other option is passing the value from make to bash, from bash to ava, and from ava to make again, but honestly, I don't have enough knowledge in js/ts to do it (and IDK if it's possible)

Reply:
So why don't we use the second approach, if it works? I mean to not introduce a new vairable `IMAGE_TAG`

Reply:
The problem is that we have 2 "make layers", the first one is the entry point, but we have another "layer" in [`setup.tests.ts` file](https://github.com/kedacore/keda/blob/main/tests/setup.test.ts#L37)
```typescript
test.serial('Deploy KEDA', t => {
  let result = sh.exec('(cd .. && make deploy)')
  if (result.code !== 0) {
    t.fail('error deploying keda. ' + result)
  }
  t.pass('KEDA deployed successfully using make deploy command')
})
```
IDK how to pass the variable to this second `make` and that's why I use environment variables. They are set and we can read them every time that we invoke `make`

Reply:
Do you mean using `VERSION = ${VERSION}` instead of `VERSION = ${IMAGE_TAG}`?
I will try, IDK if there will be conflicts but it's a good change

Reply:
It doesn't work :( 
I have tried twith this:
```make
VERSION = ${VERSION}
ifeq '$(VERSION)' ''
VERSION = main
endif
```
`Makefile:7: *** Recursive variable 'VERSION' references itself (eventually).  Stop.`
WDYT if I rename the variable to E2E_IMAGE_TAG to clarify the usage of it?

Reply:
Nice!!! I hope this afternoon I can achieve the step of deleting the tag after the process. [This Action](https://github.com/bots-house/ghcr-delete-image-action) promises

Reply:
Excellent, I am really looking forward!

Reply:
I have updated the naming to use `E2E_TEST_IMAGE_TAG`